core/src/main/java/com/digitalpebble/stormcrawler/protocol/delegate/DelegationFilters.java

```diff
@@ -0,0 +1,285 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * DigitalPebble licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.digitalpebble.stormcrawler.protocol.delegate;
+
+import com.digitalpebble.stormcrawler.Metadata;
+import com.digitalpebble.stormcrawler.protocol.Protocol;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.digitalpebble.stormcrawler.JSONResource;
+import com.digitalpebble.stormcrawler.util.ConfUtils;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang.StringUtils;
+import org.apache.storm.Config;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Map;
+
+public class DelegationFilters implements JSONResource {
+    protected String configFile;
+    protected Config stormConf;
+
+    protected Protocol robotsProtocol;
+    protected Protocol defaultProtocol;
+    protected ArrayList<JsonNode> protocolFilters;
+    protected ArrayList<Protocol> protocolImplementations;
+
+    private static final org.slf4j.Logger LOG = LoggerFactory.getLogger(DelegationFilters.class);
+
+    /**
+     * Loads and configure the DelegationFilters based on the storm config.
+     **/
+    public static DelegationFilters fromConf(Config stormConf) {
```

Comment:
not sure we need a separate class for that - the logic of loading the sub-protocols could be handled in DelegatorProtocol